core/src/main/java/jenkins/triggers/ReverseBuildTrigger.java

```diff
@@ -112,29 +113,47 @@ private boolean shouldTrigger(Run upstreamBuild, TaskListener listener) {
         if (job == null) {
             return false;
         }
+
+        boolean downstreamVisible = false;
+        boolean downstreamDiscoverable = false;
+
         // This checks Item.READ also on parent folders; note we are checking as the upstream auth currently:
-        boolean downstreamVisible = jenkins.getItemByFullName(job.getFullName()) == job;
+        try {
+            downstreamVisible = jenkins.getItemByFullName(job.getFullName()) == job;
```

Comment:
Because it will be null if the upstream auth lacks `READ` on the downstream job.